### PR TITLE
make cql code async, return future

### PR DESCRIPTION
**Pull request by danchia**
also fixes flakey test by properly waiting on store operations to return
in future.

### CODE REVIEW

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/util/FutureUtil.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.twitter.zipkin.util
+
+import com.google.common.util.concurrent.FutureCallback
+import com.google.common.util.concurrent.Futures
+import com.google.common.util.concurrent.ListenableFuture
+import com.twitter.util.Future
+import com.twitter.util.Promise
+
+object FutureUtil {
```
Comment (codefromthecrypt) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/util/FutureUtil.scala:9]: yeah this is simpler than depending on the bijection library

**Comment by codefromthecrypt** [comment]
@michaelsembwever I'd really like to merge this or some revision of it today. These build failures in cassandra are killing productivity

File: zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java [hunk 2]
```diff
@@ -195,15 +200,16 @@ public Repository(String keyspace, Cluster cluster) {
...
-                    + "if this happens a lot consider switching back to SizeTieredCompactionStrategy for "
-                    + KEYSPACE + ".traces");
-        }
+
         try {
+            if (0 == timestamp && metadata.get("traces.compaction.class").contains("DateTieredCompactionStrategy")) {
+                LOG.warn("span with no first or last timestamp. "
+                        + "if this happens a lot consider switching back to SizeTieredCompactionStrategy for "
+                        + KEYSPACE + ".traces");
+            }
```
Comment (michaelsembwever) [inline zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java:212]: why did this get moved into the try-catch ?

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala [hunk 3]
```diff
@@ -94,91 +93,106 @@ class CassandraSpanStore(
...
+    FutureUtil.toFuture(repository.getSpansByTraceIds(traceIds.toArray.map(Long.box), count))
+      .map { spansByTraceId =>
+        val spans =
+          spansByTraceId.asScala.mapValues { spans => spans.asScala.map(spanCodec.decode(_).toSpan) }
 
-      traceIds.map(traceId => spans.get(traceId)).flatten
-        .sortBy(_.head) // CQL doesn't allow order by with an "in" query
-    }
+        traceIds.flatMap(traceId => spans.get(traceId))
+          .sortBy(_.head) // CQL doesn't allow order by with an "in" query
```
Comment (michaelsembwever) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala:194]: not related to this PR, but is this trailing comment still applicable after https://github.com/openzipkin/zipkin/pull/742 got merged?

**Comment by michaelsembwever** [comment]
@adriancole i've eyeballed the PR and it looks good (fantastic).

i would have liked to have done some manual testing on it as well though.
maybe somewhere else has done this?
otherwise it's a +1 from me.

**Comment by codefromthecrypt** [comment]
@michaelsembwever by manual testing, you mean bring up the stack and click around/make traces? I can do that

**Comment by michaelsembwever** [comment]
@adriancole 

> by manual testing, you mean bring up the stack and click around/make traces? 

yes that's all.
 (stress-testing with a modified tracegen might also be cool, but something for another day i think)

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala (hunk 3 above)
Comment (codefromthecrypt) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala:194]: we can try :)

**Comment by codefromthecrypt** [comment]
ok ran collector, query, tracegen, web... all with finagle scribe logging.

File: zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java (hunk 2 above)
Comment (danchia) [inline zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java:212]: woops, no real reason for this one

**Pull request merged**